core-tests/src/test/java/org/mule/runtime/core/internal/processor/rector/profiling/ReactorProfilingUtils.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.processor.rector.profiling;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import org.mule.runtime.api.profiling.ProfilingDataProducer;
+import org.mule.runtime.core.api.event.CoreEvent;
+import org.mule.runtime.core.internal.profiling.CoreProfilingService;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.function.Consumer;
+
+/**
+ * Utils for test reactor profiling
+ */
+public class ReactorProfilingUtils {
```

Comment:
is this togglz related?

Reply:
moved to ReactorProfilingServiceTestUtils.

Reply:
This is a test utils class for mocking the reactor chain for profiling. This is related not to togglz, but now that we refactored the feature flagging service, there is a way to not always assembly the reactor chain for the processing strategy with the profiling onNext. See CoreProfilingService which is an internal extension of the profiling service to achieve this.